scipy/sparse/linalg/propack.py

```diff
@@ -0,0 +1,336 @@
+         delta=None, eta=None, anorm=0, cgs=False, elr=True, blocksize=16,
+         min_relgap=0.002, shifts=None, maxiter=None):
+    """
+    Compute the singular value decomposition of a linear operator using PROPACK
+
+    Parameters
+    ----------
+    A : array_like, sparse matrix, or LinearOperator
+        Operator for which SVD will be computed.  If `A` is a LinearOperator
+        object, it must define both ``matvec`` and ``rmatvec`` methods.
+    k : int
+        Number of singular values/vectors to compute
+    which : {"LM", "SM"}
+        Which singluar triplets to compute:
+        - 'LM': compute triplets corresponding to the `k` largest singular
+                values
+        - 'SM': compute triplets corresponding to the `k` smallest singular
+                values
+        `which='SM'` requires `irl_mode=True`.  Computes largest singular
+        values by default.
+    irl_mode : bool, optional
+        If `True`, then compute SVD using IRL (implicitly restarted Lanczos)
+        mode.  Default is `False`.
+    kmax : int, optional
+        Maximal number of iterations / maximal dimension of Krylov subspace.
+        Default is ``5 * k``.
+    compute_u : bool, optional
+        If `True` (default) then compute left singular vectors, `u`.
+    compute_v : bool, optional
+        If `True` (default) then compute right singular vectors, `v`.
+    tol : float, optional
+        The desired relative accuracy for computed singular values.
+        If not specified, it will be set based on machine precision.
+    v0 : array_like, optional
+        Starting vector for iterations: must be of length ``A.shape[0]``.
+        If not specified, PROPACK will generate a starting vector.
+    full_output : bool, optional
+        If `True`, then return sigma_bound.  Default is `False`.
+    delta : float, optional
+        Level of orthogonality to maintain between Lanczos vectors.
+        Default is set based on machine precision.
+    eta : float, optional
+        Orthogonality cutoff.  During reorthogonalization, vectors with
+        component larger than `eta` along the Lanczos vector will be purged.
+        Default is set based on machine precision.
+    anorm : float, optional
+        Estimate of ``||A||``.  Default is `0`.
+    cgs : bool, optional
+        If `True`, reorthogonalization is done using classical Gram-Schmidt.
+        If `False` (default), it is done using modified Gram-Schmidt.
+    elr : bool, optional
+        If `True` (default), then extended local orthogonality is enforced
+        when obtaining singular vectors.
+    blocksize : int, optional
+        If computing `u` or `v` (left and right singular vectors,
+        respectively), `blocksize` controls how large a fraction of the
+        work is done via fast BLAS level 3 operations.  A larger blocksize
+        may lead to faster computation at the expense of greater memory
+        consumption.  `blocksize` must be ``>= 1``.  Default is `16`.
```

Comment:
```suggestion
        consumption.  `blocksize` must be ``>= 2``.  Default is `16`.
```
(According to the code.)

I imagine the PROPACK documentation is no more specific about this? If that's the case, should we even expose the option? As a user, I wouldn't really know what this knob is doing except that a higher number might make things faster?

Reply:
I don't remember if I wrote this or it was pre-existing in pypropack, but a comment in the source code indicates:
```
    # BLAS-3 blocking size is 16, but docs don't specify; it's almost surely a
    # power of 2.
```

Probably best to remove this option and we can add back in if a reviewer complains

Reply:
Removed option, hard-coded to `16`

Reply:
Sounds good.